libraries/Microsoft.Bot.Builder.AI.LUIS/LuisV3PredictionOptions.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.AI.Luis
+{
+    /// <summary>
+    /// Optional parameters for a LUIS prediction request.
+    /// </summary>
+    public class LuisV3PredictionOptions
+    {
+        /// <summary>
+        /// Gets or sets whether all intents come back or only the top one.
+        /// </summary>
+        /// <value>
+        /// True for returning all intents.
+        /// </value>
+        public bool? IncludeAllIntents { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not instance data should be included in response.
+        /// </summary>
+        /// <value>
+        /// A value indicating whether or not instance data should be included in response.
+        /// </value>
+        public bool? IncludeInstanceData { get; set; }
+
+        /// <summary>
+        /// Gets or sets if queries should be logged in LUIS.
+        /// </summary>
+        /// <value>
+        /// If queries should be logged in LUIS.
+        /// </value>
+        public bool? Log { get; set; }
+
+        /// <summary>
+        /// Gets or sets the time in milliseconds to wait before the request times out.
+        /// </summary>
+        /// <value>
+        /// The time in milliseconds to wait before the request times out. Default is 100000 milliseconds.
+        /// </value>
+        /// <remarks>
+        /// This value can only be set when <see cref="LuisRecognizer"/> is created and can't be changed
+        /// in individual <see cref="IRecognizer.RecognizeAsync"/> calls.
+        /// </remarks>
+        public double Timeout { get; set; } = 100000;
+
+        /// <summary>
+        /// Gets or sets the datetime used for resolving relative datetime references.
+        /// </summary>
+        /// <value>
+        /// The datetime to use for resolving relative datetime references.
+        /// </value>
+        public DateTime? DatetimeReference { get; set; }
+
+        /// <summary>
+        /// Gets or sets dynamic lists used to recognize entities for a particular query.
+        /// </summary>
+        /// <value>
+        /// Dynamic lists of things like contact names to recognize at query time.
+        /// </value>
+        public IList<DynamicList> DynamicLists { get; set; }
+
```

Comment:
Consider defaulting to: = new List<DynamicList>(). This will make programming easier as devs don't need to worry about Null.

Reply:
They don't need to worry already--if it is null, it is not sent to the API.  Not even sure the API can deal with an empty list.

Reply:
I'm leaving this as is--this is really input to the prediction so people will create it and pass it in.  The get is solely for informational purposes.